### PR TITLE
lndclient: add lookup invoice

### PR DESCRIPTION
**Pull request by carlaKC**

### CODE REVIEW

File: lndclient/lightning_client.go [hunk 1]
```diff
@@ -38,6 +39,9 @@ type LightningClient interface {
 	AddInvoice(ctx context.Context, in *invoicesrpc.AddInvoiceData) (
 		lntypes.Hash, string, error)
 
+	// LookupInvoice looks up an invoice.
```
Comment (guggero) [inline lndclient/lightning_client.go:42]: nit: ...looks up an invoice by its hash. ?

File: lndclient/lightning_client.go [hunk 2]
```diff
@@ -365,6 +369,103 @@ func (s *lightningClient) AddInvoice(ctx context.Context,
 	return hash, resp.PaymentRequest, nil
 }
 
+// Invoice represents an invoice in lnd.
+type Invoice struct {
+	// Preimage is the invoice's preimage, which is set if the invoice
+	// is settled.
+	Preiamge *lntypes.Preimage
```
Comment (guggero) [inline lndclient/lightning_client.go:376]: nit: typo

File: lndclient/lightning_client.go [hunk 3]
```diff
@@ -365,6 +366,103 @@ func (s *lightningClient) AddInvoice(ctx context.Context,
 	return hash, resp.PaymentRequest, nil
 }
 
+// Invoice represents an invoice in lnd.
+type Invoice struct {
+	// Preimage is the invoice's preimage, which is set if the invoice
+	// is settled.
+	Preimage *lntypes.Preimage
+
+	// Hash is invoice hash.
```
Comment (bhandras) [inline lndclient/lightning_client.go:378]: nit: the

File: test/lightning_client_mock.go [hunk 4]
```diff
@@ -125,9 +126,38 @@ func (h *mockLightningClient) AddInvoice(ctx context.Context,
 		return lntypes.Hash{}, "", err
 	}
 
+	// Add the invoice we have created to our mock's set of invoices.
+	h.lnd.Invoices[hash] = &lndclient.Invoice{
+		Preimage:       nil,
+		Hash:           hash,
+		PaymentRequest: payReqString,
+		Amount:         in.Value,
+		CreationDate:   time.Now(),
```
Comment (bhandras) [inline test/lightning_client_mock.go:135]: nit: you could just fill `creationDate` here as it already is set to `time.Now()` earlier (also it's probably better to use that).

**Pull request merged**